### DISCUSSION TITLE
API design of checks in bundler level

### DISCUSSION THREAD

**Discussion by hyfdev** [rolldown/rolldown#2120] [Core Minutes]
```ts
defineConfig({
    checks: {
        level: 'warn', 
        rules: {
            circularReferences: 'warn', // default to `allow`
            eval: 'deny', // default to `warn`
        }
    },
})
```

```ts
defineConfig({
    clippy: {
        level: 'warn', 
        rules: {
            circularReferences: 'warn', // default to `allow`
            eval: 'deny', // default to `warn`
        }
    },
})
```

---

```ts
defineConfig({
    checks: {
      circularReferences: true,
      eval: false,
    },
})
```

**Comment by 7086cmd** [comment]
It is possible that many users may not be familiar with Rust, which could make the name `clippy` somewhat unfamiliar to them. I would suggest using the term `checks` instead, or integrating with `oxlint` for `lint` field.

**Comment by Boshen** [comment]
I'd go for `check`, to disambiguate from `oxlint`, which may be integrated at a higher level.

**Comment by hyfdev** [comment]
I'll use `checks` first. Concept like `level` would not be implemented. There will just be plain rules.
